plugin/sampling/strategystore/factory_config.go

```diff
@@ -16,32 +16,60 @@ package strategystore
 
 import (
 	"fmt"
+	"io"
 	"os"
 )
 
 const (
 	// SamplingTypeEnvVar is the name of the env var that defines the type of sampling strategy store used.
-	SamplingTypeEnvVar = "SAMPLING_TYPE"
+	SamplingTypeEnvVar = "SAMPLING_CONFIG_TYPE"
+
+	// previously the SAMPLING_TYPE env var was used for configuration, continue to support this old env var with warnings
+	deprecatedSamplingTypeEnvVar = "SAMPLING_TYPE"
+	// static is the old name for "file". we will translate from the deprecated to current name here. all other code will expect "file"
+	deprecatedSamplingTypeStatic = "static"
 )
 
 // FactoryConfig tells the Factory what sampling type it needs to create.
 type FactoryConfig struct {
 	StrategyStoreType Kind
 }
 
-// FactoryConfigFromEnv reads the desired sampling type from the SAMPLING_TYPE environment variable. Allowed values:
-//   * `static` - built-in
+// FactoryConfigFromEnv reads the desired sampling type from the SAMPLING_CONFIG_TYPE environment variable. Allowed values:
+//   * `file` - built-in
 //   * `adaptive` - built-in
-func FactoryConfigFromEnv() (*FactoryConfig, error) {
-	strategyStoreType := os.Getenv(SamplingTypeEnvVar)
-	if strategyStoreType == "" {
-		strategyStoreType = samplingTypeStatic
+func FactoryConfigFromEnv(log io.Writer) (*FactoryConfig, error) {
+	strategyStoreType := getStrategyStoreTypeFromEnv(log)
+	if strategyStoreType != samplingTypeAdaptive &&
+		strategyStoreType != samplingTypeFile &&
+		strategyStoreType != deprecatedSamplingTypeStatic {
```

Comment:
it might be even cleaner if you check for and override `static` in `getStrategyStoreTypeFromEnv`, because there was no release where `SAMPLING_CONFIG_TYPE=static` was valid.